test/controllers/actions/IdentifierActionSpec.scala

```diff
@@ -261,13 +272,385 @@ class IdentifierActionSpec
+      ))
+      when(mockSessionDataCacheConnector.fetch(ArgumentMatchers.eq(externalId))(any(), any()))
+        .thenReturn(Future.successful(Some(jsonAOP(Administrator))))
+      val controller = new Harness(authAction)
+      val enrolments = Enrolments(Set(
+        Enrolment(psaEnrolmentKey, Seq(EnrolmentIdentifier("PSAID", psaId)), "Activated", None),
+        Enrolment(pspEnrolmentKey, Seq(EnrolmentIdentifier("PSPID", pspId)), "Activated", None)
+      ))
+
+      val pstrInDB = "456"
+      val pstrJson = Json.obj(
+        "eventReporting" -> Json.obj(
+          "pstr" -> pstrInDB,
+          "schemeName" -> "schemeName",
+          "returnUrl" -> "returnUrl",
+          "srn" -> "srn"
+        )
+      )
+
+      when(mockSessionDataCacheConnector.fetch(ArgumentMatchers.eq(SessionKeys.sessionId))(any(), any()))
+        .thenReturn(Future.successful(Some(pstrJson)))
+
+      when(authConnector.authorise[Option[String] ~ Enrolments](any(), any())(any(), any()))
+        .thenReturn(Future.successful(new ~(Some("id"), enrolments)))
+
+      val result = controller.onPageLoad()(fakeRequest)
+      status(result) mustBe OK
+    }
+    "with PSP when both enrolments are available" in {
+      when(mockSchemeConnector.getSchemeDetails(any(), any(), any())(any(), any())).thenReturn(Future.successful(
+        PsaSchemeDetails("test scheme", "test pstr", "test status", None))
+      )
+      when(mockSchemeConnector.getPspSchemeDetails(any(), any())(any(), any())).thenReturn(Future.successful(
+        PspSchemeDetails("schemeName", "87219363YN", "Open", Some(PspDetails(None, None, None, psaId, AuthorisingPSA(None, None, None, None), LocalDate.now(), pspId)))
+      ))
+      when(mockSessionDataCacheConnector.fetch(ArgumentMatchers.eq(externalId))(any(), any()))
+        .thenReturn(Future.successful(Some(jsonAOP(Administrator))))
+      val controller = new Harness(authAction)
+      val enrolments = Enrolments(Set(
+        Enrolment(psaEnrolmentKey, Seq(EnrolmentIdentifier("PSAID", psaId)), "Activated", None),
+        Enrolment(pspEnrolmentKey, Seq(EnrolmentIdentifier("PSPID", pspId)), "Activated", None)
+      ))
+
+      val pstrInDB = "456"
+      val pstrJson = Json.obj(
+        "eventReporting" -> Json.obj(
+          "pstr" -> pstrInDB,
+          "schemeName" -> "schemeName",
+          "returnUrl" -> "returnUrl",
+          "srn" -> "srn"
+        )
+      )
+
+      when(mockSessionDataCacheConnector.fetch(ArgumentMatchers.eq(SessionKeys.sessionId))(any(), any()))
+        .thenReturn(Future.successful(Some(pstrJson)))
+
+      when(authConnector.authorise[Option[String] ~ Enrolments](any(), any())(any(), any()))
+        .thenReturn(Future.successful(new ~(Some("id"), enrolments)))
+
+      val result = controller.onPageLoad()(fakeRequest)
+      status(result) mustBe OK
+    }
+  }
+
+  "must be dismissed when scheme is not associated with" - {
+    "PSA" in {
```

Comment:
Is this test right? 

Looks like its the PSP that is authorised. If we change it we can remove the mock for getPspSchemeDetails?

Reply:
Removed the extra mock from PSA test.